core/Network/TLS/Crypto/DH.hs

```diff
@@ -42,7 +43,12 @@ dhGenerateKeyPair params = do
     return (priv, pub)
 
 dhGetShared :: DHParams -> DHPrivate -> DHPublic -> DHKey
-dhGetShared params priv pub = DH.getShared params priv pub
+dhGetShared params priv pub =
+    stripLeadingZeros (DH.getShared params priv pub)
+  where
+    -- strips leading zeros from the result of DH.getShared, as required
+    -- for DH(E) premaster secret in SSL/TLS before version 1.3.
+    stripLeadingZeros (DH.SharedKey sb) = DH.SharedKey (snd $ B.span (== 0) sb)
```

Comment:
Ordinarily yes, but here it's `Data.ByteArray` from memory and there is no `dropWhile`.